src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -116,10 +140,51 @@ private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity pr
             var group = searchResponse.Entries[0]; // Get the object that was found on ldap
             var groupDN = group.DistinguishedName;
 
+            if (processedGroups.Contains(groupDN)) {
+                //No need to continue, this group was resolved before
+                return;
+            }
+
+            retrievedClaims.Add(new KeyValuePair<string, string>(principal.RoleClaimType, groupCN));
             processedGroups.Add(groupDN);
 
+            //Get the group SID
+            var groupSID = group.Attributes["objectsid"];
+            if (groupSID is not null)
+            {
+                if (groupSID.Count == 1)
+                {
+                    //For some reason it is sometimes string and sometimes byte[]
+                    //when it is returned as a string then every byte is converted to a char and simply put together as a string
+                    switch (groupSID[0])
+                    {
+                        case string groupSIDstr:
+                            ReadOnlySpan<char> groupSIDspn = groupSIDstr;
+                            Span<byte> lgroupSIDba = stackalloc byte[groupSIDspn.Length];
```

Comment:
Is it possible to stack allocate to a power of two (and potentially slice then to the desired size). Perf-wise that's better, but I don't know if perf matters here that much...